falcon_mcp/common/errors.py

```diff
@@ -116,6 +116,16 @@ def handle_api_response(
     """
     status_code = response.get("status_code")
 
+    # If no status_code is present, assume success (200) for responses with data
+    # This handles cases like NGSIEM query results that don't include status_code
+    if status_code is None:
```

Comment:
Though I'm not fully familiar with the NG-SIEM API, this looks odd. Any HTTP response should contain a status line as per the RFC. In which cases have you noticed that this is `none`?